### PR TITLE
Add selection feature for dynamically selecting location format

### LINKED ISSUE #3085: Add functionality for dynamically switching between coordinate formats in location form
**Technical debt description**
When entering coordinates while editing/creating a location, provide functionality to the user for dynamically switching between respective form fields for different coordinate formats.

**Proposed solution**
Instead of showing a label on the left side of the form field we can display a dropdown. In the dropdown we display default/current coordinate format. If user selects a different format from the dropdown, corresponding form field(s) gets displayed. 

**Additional context**
See #244 #3042 and #3062

### PR DESCRIPTION
**Pull request by cemalettin-work**
Users now can change the location format when displaying or editing locations. The dictionary defines the default format (**MGRS** or **LAT_LON**). The location widget now has a selection input for dynamically changing the location format.

Closes #3085

#### User changes
- Users can change the location format between **MGRS** and **LAT_LON** on a location page.

#### Super User changes
- None

#### Admin changes
- None

#### System admin changes
- [x] anet.yml or anet-dictionary.yml needs change
  Define the default location format in the dictionary to be either **MGRS** or **LAT_LON**, e.g.:
  ```
  fields:
    location:
      format: MGRS
  ```
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
- [x] Described the user behavior in PR body
- [x] Referenced/updated all related issues
- [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
- [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
- [ ] Added and/or updated unit tests
- [x] Added and/or updated e2e tests
- [ ] Added and/or updated data migrations
- [ ] Updated documentation
- [x] Resolved all build errors and warnings
- [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: anet-dictionary.yml [hunk 1]
```diff
@@ -472,9 +472,6 @@ dictionary:
               helpText: Help text for object date field
               visibleWhen: $[?(@.colourOptions === 'GREEN')]
 
-    location:
-      format: LAT_LON
```
Comment (VassilIordanov) [inline anet-dictionary.yml:476]: We still want to specify the default format in the dictionary. I suggest to leave it here and use it as the default.

**Review by VassilIordanov — DISMISSED** [review]

File: anet-dictionary.yml (hunk 1 above)
Comment (cemalettin-simsoft) [reply anet-dictionary.yml:476]: Hmm, if we are going to set a default there, we should also set all the possible formats as array of strings. Let me commit a poc for that and we can discuss on the implementation

Comment (cemalettin-simsoft) [reply anet-dictionary.yml:476]: Now that i think about it more, defining array of formats in anet-dictionary.yml is probably useless. So one default format from the dictionary.

Comment (cemalettin-simsoft) [reply anet-dictionary.yml:476]: Another question: When a user refreshes the webpage, even if he/she selected a format, the defaulFormat will be used. Should we maybe use local storage for keeping the format setting same between refreshes?

Comment (VassilIordanov) [reply anet-dictionary.yml:476]: > When a user refreshes the webpage
it should come back to the default.
We do not have user-specific settings and we do not want to change anything globally.

Actually the original request was to have the selection of coordinate type be a part of the coordinate widget itself. I think it is cleaner and more clear. Such as:

coordinate: 18SUJ7082315291 [MGRS] 

The last part being a dropdown on how to display the coordinates

Comment (VassilIordanov) [reply anet-dictionary.yml:476]: I think it is much simpler if it is self-contained in the coordinate widget

Comment (cemalettin-simsoft) [reply anet-dictionary.yml:476]: Yes, i will include it to the location widget so we don't have to call it from multiple pages

File: anet-dictionary.yml [hunk 2]
```diff
@@ -473,7 +473,7 @@ dictionary:
               visibleWhen: $[?(@.colourOptions === 'GREEN')]
 
     location:
-      format: LAT_LON
```
Comment (VassilIordanov) [inline anet-dictionary.yml:476]: There is still a need for selecting a format for ANET. I suggest to keep the `format` field name

**Review by ysf-simsoft — CHANGES_REQUESTED** [review]

File: client/src/pages/locations/GeoLocation.js [hunk 3]
```diff
@@ -182,7 +194,9 @@ MGRSFormField.propTypes = {
   editable: PropTypes.bool,
   setFieldValue: fnRequiredWhenEditable,
   setFieldTouched: fnRequiredWhenEditable,
-  isSubmitting: PropTypes.bool
+  isSubmitting: PropTypes.bool,
+  locationFormat: PropTypes.string,
+  setLocationFormat: PropTypes.func
```
[the remark this answers is not in the document]
Comment (cemalettin-simsoft) [reply client/src/pages/locations/GeoLocation.js:199]: We are always showing the selection widget, editable or not, submitting or not.

File: client/src/pages/locations/GeoLocation.js [hunk 4]
```diff
@@ -284,7 +302,9 @@ LatLonFormField.propTypes = {
...
```
[the remark this answers is not in the document]
Comment (cemalettin-simsoft) [reply client/src/pages/locations/GeoLocation.js:307]: We are always showing the selection widget, editable or not, submitting or not.

File: client/src/components/SelectLocationFormat.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+SelectLocationFormat.propTypes = {
+  locationFormat: PropTypes.string.isRequired,
+  setLocationFormat: PropTypes.func.isRequired
+}
+
+const LocationFormatSelector = styled.select`
+  margin-left: 10px;
+  outline: 2px solid #337ab7;
+  font-size: 14px;
+  margin-left: 1rem;
```
Comment (ysf-simsoft) [inline client/src/components/SelectLocationFormat.js:34]: `margin-left` is set twice. Also I suggest removing `outline` and add `padding: 4px` which creates a better view in both Chrome and Firefox

**Review by ysf-simsoft — CHANGES_REQUESTED** [review]

File: client/src/models/Location.js [hunk 6]
```diff
@@ -17,6 +17,13 @@ export default class Location extends Model {
     REPORT_APPROVAL: "REPORT_APPROVAL"
   }
 
+  static LOCATION_FORMATS = {
+    LAT_LON: "LAT_LON", // default
+    MGRS: "MGRS"
+  }
+
+  static locationFormat = Settings.fields.location.format // Default
```
Comment (gjvoosten) [inline client/src/models/Location.js:25]: Suggested change:
```
  static LOCATION_FORMATS = {
    LAT_LON: "LAT_LON",
    MGRS: "MGRS"
  }

  static locationFormat =
    Settings.fields.location.format || Location.LOCATION_FORMATS.LAT_LON
```

File: src/main/resources/anet-schema.yml [hunk 7]
```diff
@@ -520,10 +520,11 @@ properties:
         properties:
           format:
             type: string
+            title: Default format for location
             enum: [LAT_LON, MGRS]
-            title: Coordinate format for location
             description: Used in the UI where a location's coordinate is shown. Defaults to LAT_LON.
 
+
```
Comment (gjvoosten) [inline src/main/resources/anet-schema.yml:527]: Spurious blank line.

**Review by gjvoosten — DISMISSED** [review]

**Review by anilsonmez-simsoft — APPROVED** [review]

**Pull request merged**